packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3952,6 +3968,22 @@ class InheritedElement extends ProxyElement {
   /// by first obtaining their [InheritedElement] using
   /// [BuildContext.ancestorInheritedElementForWidgetOfExactType].
   void dispatchDidChangeDependencies() {
+    assert(() {
```

Comment:
I think factoring this closure out as _debugCheckCurrentBuildTarget() or something would be better than repeating it wholesale.